services/httpd/vhosts.nix

```diff
@@ -3,287 +3,251 @@ with (import ./shared.nix { tld = config.redbrick.tld; });
 let
   users = import ./users.nix;
 
-  # This is appended at the bottom
+  # Reserved or ignored names, e.g. for service accounts
+  # where the service has its own vhost
+  userBlacklist = [
+    "paste"
+    "wiki"
+    "cmtwiki"
+  ];
+
+  # This is appended at the top
   # to ensure that custom vhosts take preference
-  userVhosts = builtins.map (user: vhost {
-    hostName = "${user.uid}.${tld}";
-    documentRoot = common.userWebtree user.uid;
-    user = user.uid;
-    group = user.gid;
-    serverAliases = [];
-  }) users;
-in (if (config.redbrick.skipVhosts) then [] else ([
-  (vhost {
-    hostName = "abovethefold.es";
+  userVhosts = with builtins; listToAttrs (map (user: {
+    name = "${user.uid}.${tld}";
+    value = vhost {
+      documentRoot = common.userWebtree user.uid;
+      user = user.uid;
+      group = user.gid;
+    };
+  }) (filter (user: !elem user.uid userBlacklist) users));
+in (if (config.redbrick.skipVhosts) then {} else userVhosts // {
+  "abovethefold.es" = vhost {
     documentRoot = "${webtree}/r/receive/abovethefold";
     user = "receive";
     group = "staff";
-  })
-  (vhost {
-    hostName = "blog.${tld}";
+    wwwRedirect = true;
+    serverAliases = [ "www.abovethefold.es" ];
+  };
+  "blog.${tld}" = vhost {
```

Comment:
so I think it'd be a good idea if we alphabetise this list. https://github.com/redbrick/nix-configs/blob/blog/services/httpd/vhosts.nix <- was done in the testing blog so the only difference is the lack of blog

Reply:
Yeah fair enough. Shall we not wait until the blog pr is merged then?

Reply:
sure it can wait till #38